### PR TITLE
Init commit

### PR DESCRIPTION
**Pull request by klzns**
#### What does this PR do? \*

Implements Product Specifications blocks
https://vtex-dev.atlassian.net/jira/software/projects/SF/boards/105?selectedIssue=SF-250

#### How to test it? \*

https://prodspec--storecomponents.myvtex.com/classic-shoes/p

image

Code added to the theme:
https://github.com/vtex-apps/store-theme/compare/master..feature/product-specifications

#### Describe alternatives you've considered, if any. \*

n/a

#### Related to / Depends on \*

n/a

### CODE REVIEW

File: react/ProductSpecification.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import React, { FC, useContext } from 'react'
+import { Specification } from 'vtex.product-context'
```
Comment (victorhmp) [inline react/ProductSpecification.tsx:2]: Nice to see this working 😄

Comment (klzns) [reply react/ProductSpecification.tsx:2]: It doesn't! I created a typing in this project haha

File: docs/README.md [hunk 2]
```diff
@@ -1,91 +1,192 @@
...
+```
+
+Now, you can use all the blocks exported by the `product-specifications` app. Check out the full list below:
+
+| Block name                    | Description                                                                                                     |
+| ----------------------------- | --------------------------------------------------------------------------------------------------------------- |
+| `product-specification-group` | Renders its children for every product specification group.                                                     |
+| `product-specification`       | Renders its children for every product specification. It should be placed bellow `product-specification-group`. |
+| `product-specification-value` | Renders its children for every product value. It should be placed bellow `product-specification`.               |
+| `product-specification-text`  | Renders the data of a specification group, specification, and specification value.                              |
```
Comment (guerreirobeatriz) [inline docs/README.md:34]: I didn't understand the blocks' description (therefore I was not able to give more details on step 2). 

In my opinion, the current table is too focused on explaining the relationship between blocks. I mean, it is important to do that, but I think the focus here should be to explain the block functionality so users can properly build the template in the next step.

Comment (klzns) [reply docs/README.md:34]: @guerreirobeatriz 

A Product Specification it's structured like this:

- Product
  - List of Specifications Groups ¹
    - Group
      - Group Name ²
      - List of Specifications ¹
        - Specification
          - Specification Name ²
          - List of Specification Values ¹
            - Specification Value ²

This app iterate through each list (¹) to render it's values (²) with the block `product-specification-text`

File: react/ProductSpecificationGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  const specificationGroups = product?.specificationGroups ?? []
+
+  const groups = useMemo(
+    () =>
+      specificationGroups.filter((group) => {
+        if (group.originalName === 'allSpecifications') {
+          return false
+        }
+
+        const haveGroup = filterSpecificationGroups.includes(group.originalName)
```
Comment (victorhmp) [inline react/ProductSpecificationGroup.tsx:33]: ⛏️ Nitpick
```suggestion
        const hasGroup = filterSpecificationGroups.includes(group.originalName)
```

File: react/ProductSpecificationText.test.tsx [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+  expect(queryByText('Specification 1')).not.toBeInTheDocument()
+  expect(queryByText('Value 1')).not.toBeInTheDocument()
+
+  rerender(
+    <ProductSpecificationGroup>
+      <ProductSpecification>
+        <ProductSpecificationText message="{groupName} {specificationName} {specificationValue}" />
+      </ProductSpecification>
+    </ProductSpecificationGroup>
+  )
```
Comment (kaisermann) [inline react/ProductSpecificationText.test.tsx:47]: Would it make sense to move these to a utility method? Tests would be a bit leaner

**Pull request merged**